frontend/src/metabase/visualizations/components/Visualization/Visualization.jsx

```diff
@@ -72,39 +72,87 @@ const mapStateToProps = state => ({
 
 const SMALL_CARD_WIDTH_THRESHOLD = 150;
 
-class Visualization extends PureComponent {
-  state = {
-    getHref: undefined,
-    hovered: null,
-    clicked: null,
-    error: null,
-    genericError: null,
-    warnings: [],
-    series: null,
-    visualization: null,
-    computedSettings: {},
+const isLoading = series => {
+  return !(
+    series &&
+    series.length > 0 &&
+    _.every(
+      series,
+      s => s.data || _.isObject(s.card.visualization_settings.virtual_card),
+    )
+  );
+};
+
+const deriveStateFromProps = props => {
+  const transformed = props.rawSeries
+    ? getVisualizationTransformed(extractRemappings(props.rawSeries))
+    : null;
+
+  const series = transformed?.series;
+
+  const computedSettings = !isLoading(series)
+    ? getComputedSettingsForSeries(series)
+    : {};
+
+  return {
+    series,
+    computedSettings,
+    visualization: transformed?.visualization,
   };
+};
+
+class Visualization extends PureComponent {
+  constructor(props) {
+    super(props);
 
-  UNSAFE_componentWillMount() {
-    this.transform(this.props);
+    this.state = {
+      getHref: undefined,
+      hovered: null,
+      clicked: null,
+      error: null,
+      genericError: null,
+      warnings: [],
+      series: null,
+      visualization: null,
+      computedSettings: {},
+      ...deriveStateFromProps(props),
```

Comment:
[From the doc](https://react.dev/reference/react/Component#static-getderivedstatefromprops), `getDerivedStateFromProps` should be called from the first render, so we probably don't need this since it would be overridden by `getDerivedStateFromProps` anyway.

Reply:
Good call, fixed in https://github.com/metabase/metabase/pull/52885/commits/6705241a2d77598e8ea52fa06b014e001c206732